### PR TITLE
User story 14

### PR DESCRIPTION
**Pull request by melatran**
User Story 14, User cannot log in with bad credentials

As a visitor
When I visit the login page ("/login")
And I submit invalid information
Then I am redirected to the login page
And I see a flash message that tells me that my credentials were incorrect
I am NOT told whether it was my email or password that was incorrect

All of this functionality works. And moved actions from login controller to sessions.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+  end
+
+  def current_merchant?
+    current_user && current_user.merchant?
+  end
+
+  def current_admin?
+    current_user && current_user.admin?
+  end
```
Comment (Rostammahabadi) [inline app/controllers/application_controller.rb:20]: great helper methods

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  def type_of_login(user)
+    if user.default?
+      redirect_to '/profile'
+    elsif user.merchant?
+      redirect_to '/merchant/dashboard'
+    elsif user.admin?
+      redirect_to '/admin/dashboard'
+    end
+  end
```
Comment (Rostammahabadi) [inline app/controllers/sessions_controller.rb:28]: such wow, much yes

File: app/views/sessions/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    <input type="email" name="email" class="form-control" id="username" placeholder="Email" required name=un>
+  </div>
+  <div class="form-group col-md-4">
+    <label for="password">Password</label>
+    <input id="password" type=password required name=password class="form-control" placeholder="Password">
+  </div>
+  <%= hidden_field_tag :authenticity_token, form_authenticity_token %>
+  <input type=submit value="Login">
+</div>
+</form>
```
Comment (Rostammahabadi) [inline app/views/sessions/new.html.erb:14]: looks clean

File: spec/features/sessions/login_spec.rb [hunk 4]
```diff
@@ -78,8 +78,45 @@
...
+
+    user = User.create(name: "Natasha Romanoff",
+      address: "890 Fifth Avenue",
+      city: "Manhattan",
+      state: "New York",
+      zip: "10010",
+      email: "spiderqueen@hotmail.com",
+      password: "arrow",
+      role: 0)
+
```
Comment (JPuchalla) [inline spec/features/sessions/login_spec.rb:152]: Great tests for verification of credentials!

**Comment by Rostammahabadi** [comment]
Looks good, i approve!

**Pull request merged**